src/App.js

```diff
@@ -1,39 +1,99 @@
 import React from "react";
+import { Route, withRouter } from 'react-router-dom';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { fetchAllUsers } from './store/actions/users';
+import { loginRequest, registerUser, logout } from './store/actions/auth';
+
+import NavBar from './components/NavBar';
+import Logout from './components/Logout';
+import HomePage from './views/HomePage';
+import Chat from './views/Chat';
+import Login from './views/Login';
+import Register from './views/Register';
+import './App.css';
 
 class App extends React.Component {
   constructor(props) {
     super(props);
     this.props = props;
   }
 
-  componentWillMount() {
-    const {dispatchFetchAllUsers} = this.props;
-    dispatchFetchAllUsers();
-  }
-
   render() {
-    const { state } = this.props;
+    const { state, dispatchLoginRequest, dispatchRegisterUser, dispatchFetchAllUsers, dispatchLogout } = this.props;
     return (
-      <div className='App'>
-        <h1>Hello, World!</h1>
-        {state.users.map(user => (
-            <p key={user._id}> {user.name} </p>
-        ))}
+      <div className="App">
+        <NavBar loggedIn={Boolean(state.authToken)} />
+        <Route
+          exact
+          path="/"
+          render={props => (
+            <HomePage
+              {...props}
+              loggedIn={Boolean(state.authToken)}
+              fetchAllUsers={dispatchFetchAllUsers}
+            />
+          )}
+        />
+        <Route
+          path="/login"
```

Comment:
Do we need a separate guest login path and employee login?

Reply:
Yes, I think it will be a matter of having options to login as guest or staff on the login page and redirect as appropriate but we can decide later.

Reply:
If we do it that way I think by default the login should be for a guest with a button somewhere to go to the employee login. Although,  I think we should probably have 2 different clients. A guest client and a staff client